src/features/PassSurvey/model/ConditionalLogicBuilder.ts

```diff
@@ -0,0 +1,100 @@
+import { ConditionalLogicValidator } from './ConditionalLogicValidator';
+
+import { appletModel } from '~/entities/applet/';
+import { Condition } from '~/shared/api';
+
+type Item = appletModel.ItemRecord;
+
+export type ItemMapByName = Record<string, Item>;
+
+class ConditionalLogicBuilder {
+  private hiddenOrSkippedItemNames: Set<string> = new Set();
+  private hiddenOrSkippedItemIds: Set<string> = new Set();
```

Comment:
But how about creating an array like this:
```
private hiddenOrSkippedItems: Array<Item>
```

I understand the convenience of using Sets with primitives: it filters out duplicates automatically upon invoking "add". However, it populates the code with unnecessary lines likes this doing the same thing for the same objects:

this.hiddenOrSkippedItemNames.delete(item.name);
this.hiddenOrSkippedItemIds.delete(item.id);

Reply:
It doesn`t really help to reduce code. If we change a set to an array we should find the item each time when we want to manipulate it. 

So I don`t think that it improves readability or something. I suggest to keep it as is.